pyrs/dataobjects/fields.py

```diff
@@ -767,48 +775,44 @@ def __to_wksp_and_peaks(filename: str,
             if closeproject:
                 projectfile.close()
                 del projectfile
+
+            # verify the subruns are parallel
+            if hidraworkspace and hidraworkspace.get_sub_runs() != peak_collection.sub_runs:  # type: ignore
+                raise RuntimeError('Need to have matching subruns')
         elif TYPE_CHECKING:
             hidraworkspace = cast(HidraWorkspace, hidraworkspace)
             peak_collection = cast(PeakCollection, peak_collection)
 
+        # extract the PointList
+        if hidraworkspace:
+            if not point_list:
+                point_list = hidraworkspace.get_pointlist()
+        elif TYPE_CHECKING:
```

Comment:
a one-liner comment explaining the purpose of `TYPE_CHECKING` would be very helpful here